uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/manager-leave/components/manager-leave-calendar_test.js

```diff
@@ -1,30 +1,412 @@
+              expect(dateObj.UI.isNonWorkingDay).toBe(true);
+            });
+          });
+        });
+
+        describe('isPublicHoliday', function () {
+          var dateObj;
+
+          beforeEach(function () {
+            spyOn(controller, 'isPublicHoliday').and.returnValue(true);
+            controller.refresh();
+            $rootScope.$digest();
+          });
+
+          it('is set', function () {
+            _.each(controller.managedContacts, function (contact) {
+              dateObj = contact.calendarData.days[Object.keys(contact.calendarData.days)[0]];
+              expect(dateObj.UI.isPublicHoliday).toBe(true);
+            });
+          });
+        });
+
+        describe('refresh', function () {
+          var dateObj,
+            workPattern,
+            leaveRequest;
+
+          beforeEach(function () {
+            workPattern = workPatternData.daysData();
+            leaveRequest = leaveRequestData.singleDataSuccess().values[0];
+            workPattern.values[0].date = leaveRequest.from_date;
+          });
+
+          describe('when leave request is not approved', function () {
+            beforeEach(function () {
+              var status = _.find(optionGroupMock.getCollection('hrleaveandabsences_leave_request_status'), function (status) {
+                return status.name === 'waiting_approval';
+              });
+              leaveRequest.contact_id = '203';
+              leaveRequest.status_id = status.value;
+              leaveRequest.balance_change = -1;
+              commonSetup();
+            });
+
+            it('isRequested flag is true', function () {
+              _.each(controller.managedContacts, function (contact) {
+                dateObj = contact.calendarData.days[getDateObjectWithFormat(leaveRequest.from_date).valueOf()];
+                expect(dateObj.UI.isRequested).toBe(true);
+              });
+            });
+
+            it('styles are fetched', function () {
+              var color = _.find(controller.absenceTypes, function (absenceType) {
+                return absenceType.id == leaveRequest.type_id;
+              }).color;
+
+              _.each(controller.managedContacts, function (contact) {
+                dateObj = contact.calendarData.days[getDateObjectWithFormat(leaveRequest.from_date).valueOf()];
+                expect(dateObj.UI.styles).toEqual({
+                  backgroundColor: color,
+                  borderColor: color
+                });
+              });
+            });
+          });
+
+          describe('when leave request is for half day am', function() {
+            beforeEach(function() {
+              var halfDayAMValue = _.find(optionGroupMock.getCollection('hrleaveandabsences_leave_request_day_type'), function (absenceType) {
+                return absenceType.name === 'half_day_am';
+              }).value;
```

Comment:
is there an helper method on optionGroupMock to get by value? same for other calls below